### ISSUE TITLE
Add automated UI testing to CRviz

### ISSUE THREAD

**Issue by rashley-iqt** [CyberReboot/CRviz#75]
we should implement UI level test scripts.  We just had a bug that @lilchurro found when reviewing PR #73 where updating dependencies changed visual behavior.  It was minor and easily fixed but we need to prevent this kind of thing in the future.  It looks like the default for react is a combination of Jest and Enzyme but I'll dig into it more before I implement anything.  It looks like storybook is another popular option.

**Comment by cglewis** [comment]
I'd check out selenium too.

**Comment by rashley-iqt** [comment]
selenium could be one aspect of it, but i'd also like something more granular.  Selenium is very good for saying that a particular path through an application is broken.  I'm hoping to find something that can tell up when a particular component or ui element is broken. I'd also like something that can tell us when a change messes up our css.

**Comment by cglewis** [comment]
cool, that makes sense. i was only suggesting selenium, as that mimics the user's actual experience.

**Comment by cglewis** [comment]
it's been a while since I've used selenium, but I'm pretty sure you can just specify a setup of which browsers you want to test against, and then write a single test that runs against the browsers that were setup, rather than tests specific to browsers.

**Comment by rashley-iqt** [comment]
so i looked into Jest for this and their approach is:
https://facebook.github.io/jest/blog/2016/07/27/jest-14.html

TLDR; they expect you to generate the expected html and then keep that in your repo as part of your testing.  this strikes me as a maintenance nightmare and also just kind of trivializing the whole thing.  essentially we are just printing the site out and asking if it still looks like the picture.

**Comment by cglewis** [comment]
yikes, I'd down vote that choice too.
